File: Java/백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.java
import org.w3c.dom.ls.LSOutput;

import java.math.BigInteger;
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static final int LEN = 1_000_000;
    static boolean[] sieve = new boolean[LEN + 1];

    public static void main(String[] args) throws IOException {

        int n;
        for (int i = 2; i <= LEN; i++)
            sieve[i] = true;

        // 에라토스테네스의 체 구현
        for (int i = 2; i <= (int) Math.sqrt(LEN) + 1; i++)
            if (sieve[i])
                for (int j = i * i; j <= LEN; j += i)
                    sieve[j] = false; // 합성수 제거

        while (true) {
            n = Integer.parseInt(br.readLine());
            if (n == 0)
                break;

            for (int i = 2; i <= (int) (n / 2); i++)
                if (sieve[i] && sieve[n - i]) {
                    bw.write(String.format("%d = %d + %d", n, i, n - i) + '\n');
                    break;
                }
        }

        bw.close();
    }
}

File: Python/백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.py
from sys import stdin
from math import isqrt

input = stdin.readline

LEN = 1_000_000
sieve = [False, False] + [True] * (LEN - 1)
for i in range(2, isqrt(LEN) + 1):
    if sieve[i]:
        for j in range(i * i, LEN + 1, i):
            sieve[j] = False  # 합성수 제거

while True:
    n = int(input().rstrip())
    if n == 0:
        break

    for i in range(2, n // 2 + 1):
        if sieve[i] and sieve[n - i]:
            print(f"{n} = {i} + {n - i}")
            break
